mantidproject/mantid
# SaveCanSAS and SaveAscii2 - one spectrum per file

**Description of work.**

This PR introduces a new boolean parameter for the SaveCanSAS algorithm: `OneSpectrumPerFile`.
* if `true`, the algorithm will produce one file per spectrum
* if `false` all spectra will be added to the same file in separate `<SASdata>` entries.

This seems to correspond to the CanSAS file specifications: http://www.cansas.org/formats/canSAS1d/1.1/doc/overview.html#xml-layout-for-multiple-experiments

**To test:**

* Create a sample workspace
```python
from mantid.simpleapi import *
CreateSampleWorkspace(OutputWorkspace="ws")
```
* Export with SaveCanSAS, use the OneSpectrumPerFile option to modify the behavior.
* Observe the files and try to open them with SASView or any program that is able to read cansas files (concerning SASView, the current release is not able to properly load file with multiple `<SASdata>`, the code containing this PR should be used instead: https://github.com/SasView/sasview/pull/1890)

Part of #31943 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
